float32[] desired
float32[] achived
float32[] error
